cmd/api-linter/integration_test.go

```diff
@@ -277,6 +277,131 @@ func TestMultipleFilesFromParentDir(t *testing.T) {
 	}
 }
 
+func TestImportFromAnotherRoot(t *testing.T) {
+	// This test case is based on a scenario described in:
+	// https://github.com/googleapis/api-linter/pull/1519
+	//
+	// It checks that the linter can correctly resolve imports when
+	// one import is in a directory provided by `-I` and another
+	// is relative to the working directory. i.e.
+	//
+	// .
+	// ├── api
+	// │   ├── common
+	// │   │   └── common.proto
+	// │   └── v1
+	// │       └── test.proto
+	// └── third_party
+	//     └── google
+	//         └── api
+	//             └── field_behavior.proto
+
+	projDir, err := os.MkdirTemp("", "proj")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(projDir)
+
+	// Create the subdirectory for protos.
+	apiV1Dir := filepath.Join(projDir, "api", "v1")
+	if err := os.MkdirAll(apiV1Dir, 0755); err != nil {
+		t.Fatal(err)
+	}
+	apiCommonDir := filepath.Join(projDir, "api", "common")
+	if err := os.MkdirAll(apiCommonDir, 0755); err != nil {
+		t.Fatal(err)
+	}
+	thirdPartyDir := filepath.Join(projDir, "third_party", "google", "api")
+	if err := os.MkdirAll(thirdPartyDir, 0755); err != nil {
+		t.Fatal(err)
+	}
+
+	// Write the proto files.
+	if err := writeFile(filepath.Join(apiV1Dir, "test.proto"), `
+		syntax = "proto3";
+
+		package api.v1;
+
+		import "google/api/field_behavior.proto";
+		import "api/common/common.proto";
+
+		message Test {
+		  string name = 1 [(google.api.field_behavior) = IDENTIFIER];
+
+		  api.common.Foo foo = 2;
+		}
+	`); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := writeFile(filepath.Join(apiCommonDir, "common.proto"), `
+		syntax = "proto3";
+
+		package api.common;
+
+		message Foo {}
+	`); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := writeFile(filepath.Join(thirdPartyDir, "field_behavior.proto"), `
```

Comment:
Can we make the test simpler (just import a type, not use an option) and not use a real proto (avoid using a copy of the real `google/api/field_behavior.proto`) in the test? It is unlikely, but global proto registry collisions could happen in this test given that we force register the common protos.